rest/config_startup.go

```diff
@@ -73,6 +76,20 @@ func DefaultStartupConfig(defaultLogFilePath string) StartupConfig {
 		},
 		MaxFileDescriptors: DefaultMaxFileDescriptors,
 	}
+
```

Comment:
I feel like we shouldn't be doing the memlimit/mem calls when populating the config, and insterad this should instead be done in NewServerContext when the server context is created (in the scenario where config.API.HeapProfileCollectionThreshold is not specified).  That also avoids logging here when logging isn't initialized, and also avoids this work altogether if the user's bootstrap config has explicitly set HeapProfileCollectionThreshold.